### PR TITLE
Resource endpoint uplift

### PR DESCRIPTION
**Pull request by whutomo**
**Summary of change**

  - Created "Resource" and "Resources" response models.
  - Uplifted retrieving, updating and creating a resource to implement the "Resource" response instance.
  - Uplifted retrieving list of resources to implement the "Resources" response instance.
  - Added a new header param "Location" which then gets implemented into "Resource" response header model.
  
  live preview: https://developers-next.whispirdev.com/docs/api/branches/resource-endpoint-uplift

### CODE REVIEW

File: .spectral.yaml [hunk 1]
```diff
@@ -177,7 +177,6 @@ overrides:
   - rules:
       payload-examples: off
     files:
-      - '**#/paths/~1workspaces~1{workspaceId}~1resources' # Resources Endpoint Examples
```
Comment (whutomo) [inline .spectral.yaml:180]: Removed overrides to enable spectral lint on resources endpoints

File: openapi.yaml [hunk 2]
```diff
@@ -1340,8 +1340,8 @@ paths:
 
         **mimeType:** String  
         The mime type of the file that is being uploaded. The type is dependent on the scope of the resource.
-          *   **Public -** Public files can be of any mimetype. text/html; text/csv
-          *   **Private -** Private files are restricted to CSV, JSON, and XML
+          *   **Public -** Public files can be of any mimetype. `text/html`; `text/csv`
+          *   **Private -** Private files are restricted to `CSV`, `JSON`, and `XML`
```
Comment (whutomo) [inline openapi.yaml:1344]: subtle changes to make it easier to read

File: openapi.yaml [hunk 3]
```diff
@@ -1375,59 +1375,11 @@ paths:
                   scope: private
                   mimeType: application/json
                   derefUri: eyAKICAgICJzYW1wbGUiOiB0cnVlCn0=
-          application/vnd.whispir.resource-v1+xml:
```
Comment (whutomo) [inline openapi.yaml:1378]: Removed xml version of request body

File: openapi.yaml [hunk 4]
```diff
@@ -10084,8 +9973,3 @@ components:
           - application/vnd.whispir.auth-v1+json
           - application/vnd.whispir.auth-v1+xml
       description: Application specific mime-type.
-    Location:
```
Comment (whutomo) [inline openapi.yaml:10087]: Removed duplicated "Location" headers, keeping the detailed version

**Review by richardfontein — APPROVED** [review]

**Pull request merged**